src/electionguard/decryption.py

```diff
@@ -111,17 +110,20 @@ def compute_decryption_share_for_cast_contests(
     """
     Compute the decryption for all of the cast contests in the Ciphertext Tally
     """
-    cpu_pool = Pool(cpu_count())
     contests: Dict[CONTEST_ID, CiphertextDecryptionContest] = {}
+    scheduler = Scheduler()
 
     for contest in tally.cast.values():
         selections: Dict[SELECTION_ID, CiphertextDecryptionSelection] = {}
-        selection_decryptions = cpu_pool.starmap(
+        selection_decryptions: List[
+            Optional[CiphertextDecryptionSelection]
+        ] = scheduler.schedule(
             compute_decryption_share_for_selection,
             [
                 (guardian, selection, context)
                 for (_, selection) in contest.tally_selections.items()
             ],
+            with_shared_resources=True,
```

Comment:
the `with_shared_resources` flag tells the scheduler to schedule in a way that shares state across the different iterations